### PR TITLE
Fix column remapping for TypedInstance.to_pandas()

### PR DESCRIPTION
**Pull request by erlendvollset**
## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [7.60.3] - 2024-09-14
+### Added
+- [Feature Preview - alpha] Fix bug in column name remapping for `TypedInstance.to_pandas()`
```
Comment (haakonvt) [inline CHANGELOG.md:22]:
```suggestion
- Fix bug in column name remapping for `TypedInstance.to_pandas()`
```

File: cognite/client/data_classes/data_modeling/instances.py [hunk 2]
```diff
@@ -465,7 +464,8 @@ def to_pandas(  # type: ignore [override]
             if not extra:
                 prop_df.columns = prop_df.columns.str.removeprefix("{}.{}/{}.".format(*view_id.as_tuple()))
                 if isinstance(self, TypedInstance):
-                    prop_df.rename(columns=_get_descriptor_property_name_mapping(type(self)), inplace=True)
+                    attr_name_mapping = self._get_descriptor_property_name_mapping()
```
Comment (haakonvt) [inline cognite/client/data_classes/data_modeling/instances.py:467]: If you do the remapping before `removeprefix(..)` there are no name conflicts (assuming you pad lookup with space.view/version)

File: tests/tests_unit/test_data_classes/test_data_models/test_typed_instances.py [hunk 3]
```diff
@@ -395,16 +426,12 @@ def test_cognite_asset_read_and_write__to_pandas(
...
-            ],
-            columns=["value"],
-            index=["type", "type"],
+        expected_type_series = pd.Series(
+            [{"space": "should-be", xid: "at-root"}],
+            index=["value"],
+            name="type",
         )
-        pd.testing.assert_frame_equal(read_expanded_df.loc["type"], expected_type_df)
+        pd.testing.assert_series_equal(read_expanded_df.loc["type"], expected_type_series)
```
Comment (haakonvt) [inline tests/tests_unit/test_data_classes/test_data_models/test_typed_instances.py:434]: After these changes, this is now already covered on line 427. Perhaps we add a check on `asset_type` (or what the alias name was) instead?

**Review by haakonvt — APPROVED** [review]

File: cognite/client/data_classes/data_modeling/instances.py (hunk 2 above)
Comment (erlendvollset) [reply cognite/client/data_classes/data_modeling/instances.py:467]: The renaming addresses conflicts with the base properties, not other properties within the prop object (that necessarily does not have conflicts).

File: tests/tests_unit/test_data_classes/test_data_models/test_typed_instances.py (hunk 3 above)
Comment (erlendvollset) [reply tests/tests_unit/test_data_classes/test_data_models/test_typed_instances.py]: Fixed.

File: cognite/client/data_classes/data_modeling/instances.py (hunk 2 above)
Comment (haakonvt) [reply cognite/client/data_classes/data_modeling/instances.py:467]: Oh right

**Pull request merged**